### PR TITLE
reduce middleware complexity

### LINKED ISSUE #761: cannot download a file with phoenix
1. upload a file
2. try to download the file using phoenix

Result:
no output in the browser

ocis logs:
```
2020-10-28T11:57:59+05:45 ERR could not retrieve signing key error="{\"id\":\"com.owncloud.api.store\",\"code\":404,\"detail\":\"could not read record\",\"status\":\"Not Found\"}" service=proxy

### PR DESCRIPTION
**Pull request by fschade**
* refactor middlewares and reduce technical complexity
* restructure server command and remove cfg.OIDC.Issuer switch, oidc middleware detects now if it should used or not
* fix #761

### CODE REVIEW

File: proxy/pkg/middleware/signed_url_auth.go [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+	if err != nil {
+		return true, err
+	}
+
+	expires, err := time.ParseDuration(query.Get("OC-Expires") + "s")
+	if err != nil {
+		return true, err
+	}
+
+	date.Add(expires)
```
Comment (C0rby) [inline proxy/pkg/middleware/signed_url_auth.go:187]: Oh, this should be `date = date.Add(expires)` otherwise the `Add` won't have any effect.

Comment (fschade) [reply proxy/pkg/middleware/signed_url_auth.go:187]: good point, i also wondered what magic is going on... its a copy from the old middleware...

i update it to: 
`now > date && now < expires`

Comment (C0rby) [reply proxy/pkg/middleware/signed_url_auth.go:187]: `expires` is just a duration not a time/date you need to add it to date

File: proxy/pkg/command/server.go [hunk 2]
```diff
@@ -247,73 +246,67 @@ func Server(cfg *config.Config) *cli.Command {
...
-		middleware.AccountsClient(accounts),
-		middleware.SettingsRoleService(roles),
-		middleware.AutoprovisionAccounts(cfg.AutoprovisionAccounts),
-		middleware.EnableBasicAuth(cfg.EnableBasicAuth),
-		middleware.OIDCIss(cfg.OIDC.Issuer),
-	)
-
-	// the connection will be established in a non blocking fashion
-	sc, err := cs3.GetGatewayServiceClient(cfg.Reva.Address)
+	accountsClient := acc.NewAccountsService("com.owncloud.api.accounts", mclient.DefaultClient)
```
Comment (kulmann) [inline proxy/pkg/command/server.go:249]: Afaict all occurrences of `mclient.DefaultClient` should now be replaced with `grpc.DefaultClient`, with the grpc package coming from `"github.com/owncloud/ocis/ocis-pkg/service/grpc"`, as of https://github.com/owncloud/ocis/pull/840

**Review by IljaN — APPROVED** [review]

**Pull request merged**